src/tower/_client.py

```
@@ -119,3 +119,21 @@ def wait_for_run(run: Run) -> None:
                     return
                 else:
                     time.sleep(WAIT_TIMEOUT)
+
+
+def wait_for_runs(runs: List[Run]) -> None:
+    """
+    `wait_for_runs` waits for a list of runs to reach a terminal state by
+    polling the Tower API every 2 seconds for the latest status. If any of the
+    runs return a terminal status (`exited`, `errored`, `cancelled`, or
+    `crashed`) then this function returns.
+
+    Args:
+        runs (List[Run]): A list of runs to wait for.
+
+    Raises:
+        RuntimeError: If there is an error fetching the run status or if any
+            of the runs fail.
+    """
+    for run in runs:
+        wait_for_run(run)
```

This implementation is fine for the success case (all runs that we are waiting for are successful).

But in control flows that I am familiar with (e.g. Databricks or Google Dataflow), when people are waiting for 3 child apps, if one of them ends with a failure, the entire parent pipeline fails too.

I think this implementation is ok for v1, but we should think about 
- Returning run status in the wait_for_run()
- in the wait_for_runs(), check for run status of the children, and fail the function if ANY of the child runs fails.

---

This is a really good point. What do you think the default behavior should be? I was considering adding a flag that controls the failure behavior.

```
def wait_for_runs(runs: List[Runs], raise_on_failure: bool = True) -> None:
```

And if any of these fail, we'd raise an exception (and stop the parent pipeline). Perhaps that should be the default behavior? Likewise, perhaps it should return a tuple of `(successful_runs, failed_runs)` instead of `None`? What do you think?

---

@bradhe If we look at how other orchestrators do it, they have a pretty complex framework for defining what to do in case of task failures, for a good reason (to allow users to reuse partial calculations and avoid rerunning the entire pipeline). 

Airflow:
https://medium.com/@jaywang.recsys/apache-airflow-recovery-from-a-tasks-e66234b6d671
Dagster:
https://docs.dagster.io/guides/deploy/execution/run-retries

I think it would be too early to introduce these retry mechanisms, but at least we should tell users if the child app was successful or not. Then they can decide in the orchestrator app what to do: retry or abort.

The raise_on_error where we would throw an exception would be helpful in use cases when the user wants to stop executing the parent app without dealing with child app statuses. But if we add this parameter, it needs to be set to False by default, otherwise users won't be able to deal with failures themselves.

So, I think this means, 

Option 1: the user deals with retries themselves and also decides whether to abort the parent app themselves.
def wait_for_run(run: Run) -> Bool:
   returns success or failure

def wait_for_runs(runs: List[Run]) -> (successful_runs, failed_runs):
   returns a list of successes/failures


Option 2: the user still has to deal with retries themself, but can tell us whether to abort the parent app or not.
def wait_for_run(run: Run, raise_on_failure: bool = False) -> Bool:
   returns success or failure

def wait_for_runs(runs: List[Runs], raise_on_failure: bool = False) -> (successful_runs, failed_runs):
   returns a list of successes/failures, or raises an exception on first failure among children, if user decided to do so